src/metrics.ts

```diff
@@ -63,8 +65,260 @@ function initializeMetricMaps() {
+          distanceMoved: 0,
+        },
+        'bRh': {
+          initialPosition: 'h8',
+          currentPosition: 'h8',
+          distanceMoved: 0,
+        },
+        'bPa': {
+          initialPosition: 'a7',
+          currentPosition: 'a7',
+          distanceMoved: 0,
+        },
+        'bPb': {
+          initialPosition: 'b7',
+          currentPosition: 'b7',
+          distanceMoved: 0,
+        },
+        'bPc': {
+          initialPosition: 'c7',
+          currentPosition: 'c7',
+          distanceMoved: 0,
+        },
+        'bPd': {
+          initialPosition: 'd7',
+          currentPosition: 'd7',
+          distanceMoved: 0,
+        },
+        'bPe': {
+          initialPosition: 'e7',
+          currentPosition: 'e7',
+          distanceMoved: 0,
+        },
+        'bPf': {
+          initialPosition: 'f7',
+          currentPosition: 'f7',
+          distanceMoved: 0,
+        },
+        'bPg': {
+          initialPosition: 'g7',
+          currentPosition: 'g7',
+          distanceMoved: 0,
+        },
+        'bPh': {
+          initialPosition: 'h7',
+          currentPosition: 'h7',
+          distanceMoved: 0,
+        },
+      };
+
+      // Initialize variables to keep track of the maximum distance and the piece
+      let maxDistance = -1;
+      let maxDistancePiece = null;
+
+      ingestedMoves.forEach(move => {
+        
+        const fromSquare = move.from;
+        const toSquare = move.to;
+        //console.log(`fromSquare: ${fromSquare}`);
+        //console.log(`toSquare: ${toSquare}`);
+
+
+        function calculateDistance(fromSquare: string, toSquare: string) {
+          let totalDistance;
+
+          // Calculate the distance moved by the piece
+          const fileDistance = Math.abs(fromSquare.charCodeAt(0) - toSquare.charCodeAt(0));
+          const rankDistance = Math.abs(Number(fromSquare[1]) - Number(toSquare[1]));
+
+          // Check if it's a diagonal move
+          if (fileDistance === rankDistance) {
+            // This is a diagonal move, count it as a single horizontal/vertical movement
+            totalDistance = fileDistance;
+            console.log(`Total distance moved: ${totalDistance}`);
+          } 
+          else {
+            // Not a diagonal move, calculate the total distance, where one square = 1 unit of distance
+            totalDistance = fileDistance + rankDistance;
+            console.log(`Total distance moved: ${totalDistance}`);
+          }
+          return totalDistance;
+        }
+
+        // identifying the right piece to update the current position and distance of
+        const thisPiece = chess.get(fromSquare as Square);
+        if (thisPiece && thisPiece.type) {
+          const color = thisPiece.color === 'w' ? 'w' : 'b';
+          const pieceType = thisPiece.type.toUpperCase();
+
+          // Iterate over the pieces object to find the piece with a matching currentPosition
+          for (const pieceKey in pieces) {
```

Comment:
this is suboptimal, we are checking every key (worst case) every time. We won't need to do that using the move history